pytket/pytket/_tket/circuit.pyi

```diff
@@ -3686,7 +3686,7 @@ class StabiliserAssertionBox(Op):
         """
 class StatePreparationBox(Op):
     """
-    A box for preparing quantum states using multiplexed-Ry and multiplexed-Rz gates
+    A box for preparing quantum states using multiplexed-Ry and multiplexed-Rz gates. Implementation based on Theorem 9 of https://arxiv.org/abs/quant-ph/0406176. The decomposed circuit has at most 2*(2^n-2) CX gates, and 2^n-2 CX gates if the coefficients are all real. Note that by decomposing each multiplexed-Ry(Rz) as its dagger can lead to a CX cancellation with its adjacent multiplexed-Rz(Ry), which leads to an improved CX count of 2^(n+1)-2n. However, this is currently not automatically done in tket.
```

Comment:
Implementation based on Theorem 9 of arxiv.org/abs/quant-ph/0406176. The decomposed circuit has at most 2*(2^n-2) CX gates, and 2^n-2 CX gates if the coefficients are all real. Note that by decomposing each multiplexed-Ry(Rz) as its dagger can lead to a CX cancellation with its adjacent multiplexed-Rz(Ry), which leads to an improved CX count of 2^(n+1)-2n. However, this is currently not automatically done in tket.

Reply:
I might suggest we remove the note on the improvement though, and instead add a github issue to implement the change?

Reply:
Removed. Github issue is the right place for that note.